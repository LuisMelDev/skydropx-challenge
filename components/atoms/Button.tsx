import React, { ButtonHTMLAttributes } from "react";

export const Button = ({
    children,
    type = "button",
    onChange,
    className,
}: ButtonHTMLAttributes<HTMLButtonElement>) => {
    return (
        <button type={type} onChange={onChange} className={className}>
            {children}
        </button>
    );
};
