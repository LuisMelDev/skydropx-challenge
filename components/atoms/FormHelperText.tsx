import React from "react";

interface Props {
  text: string;
  className?: string;
}

export const FormHelperText = ({ text, className = "" }: Props) => {
  return <span className={className}>{text}</span>;
};
