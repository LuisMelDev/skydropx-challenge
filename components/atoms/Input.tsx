import React, { ChangeEventHandler, HTMLInputTypeAttribute } from "react";

interface Props {
  placeholder?: string;
  onChange?: ChangeEventHandler;
  className?: string;
  id: string;
  type: HTMLInputTypeAttribute;
  value: any;
  name?: string;
}

export const Input = ({
  placeholder = "",
  onChange,
  id,
  className = "",
  type,
  value = "",
  name = "",
}: Props) => {
  return (
    <input
      id={id}
      placeholder={placeholder}
      onChange={onChange}
      className={className}
      type={type}
      value={value}
      name={name}
    />
  );
};
