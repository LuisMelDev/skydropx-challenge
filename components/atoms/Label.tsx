import React from "react";

interface Props {
  text: string;
  htmlFor: string;
  className?: string;
}

export const Label = ({ htmlFor, text, className = "" }: Props) => {
  return (
    <label htmlFor={htmlFor} className={className}>
      {text}
    </label>
  );
};
