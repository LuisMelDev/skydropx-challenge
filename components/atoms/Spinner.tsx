import React from "react";

export const Spinner = () => {
    return (
        <div className='flex justify-center items-center h-screen fixed top-0 left-0 right-0 bg-white/50 backdrop-blur-sm'>
            <div className='relative w-24 h-24 animate-spin rounded-full bg-gradient-to-r from-purple-400 via-blue-500 to-red-400 '>
                <div className='absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 w-20 h-20 bg-gray-200 rounded-full border-2 border-white'></div>
            </div>
        </div>
    );
};
