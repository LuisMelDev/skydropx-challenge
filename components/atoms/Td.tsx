import React from "react";

export const Td = ({ children }: { children: JSX.Element | null }) => {
    return (
        <td className='px-5 py-5 border-b border-gray-200 text-sm text-gray-900 whitespace-no-wrap'>
            {children}
        </td>
    );
};
