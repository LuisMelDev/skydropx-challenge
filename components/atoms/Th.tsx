import React from "react";

export const Th = ({ name }: { name: string }) => {
    return (
        <th className='px-5 py-3 text-left text-xs font-semibold text-gray-600 uppercase tracking-wider'>
            {name}
        </th>
    );
};
