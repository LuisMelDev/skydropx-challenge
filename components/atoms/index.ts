export * from "./Label"
export * from "./Input"
export * from "./Button"
export * from "./Th"
export * from "./Tab"
export * from "./Td"
export * from "./ErrorMessage"
export * from "./Spinner"