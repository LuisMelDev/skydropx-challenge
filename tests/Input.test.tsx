import React from "react";
import { shallow } from "enzyme";
import { Input } from "../components/atoms/Input";

describe("Pruebas <ErrorMessage>", () => {
    const onChange = jest.fn();
    let wrapper = shallow(
        <Input id='inputPrueba' type='text' value='' onChange={onChange} />
    );

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(
            <Input id='inputPrueba' type='text' value='' onChange={onChange} />
        );
    });

    test("debe de mostrarse correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("Debe ejecutar onchan correctamente", () => {
        const input = wrapper.find("input");
        const value = "Hola Mundo";
        const event = { target: { value } };
        input.simulate("change", event);

        expect(onChange).toHaveBeenCalledWith(event);
    });
});
