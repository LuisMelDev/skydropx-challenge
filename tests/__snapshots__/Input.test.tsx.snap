// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <ErrorMessage> debe de mostrarse correctamente 1`] = `
<input
  className=""
  id="inputPrueba"
  name=""
  onChange={[MockFunction]}
  placeholder=""
  type="text"
  value=""
/>
`;
